proposals/first-class-span-types.md

```diff
@@ -109,6 +109,55 @@ The compiler expects to use the following helpers or equivalents to implement th
 | ReadOnlySpan to ReadOnlySpan | `static ReadOnlySpan<T>.CastUp<TDerived>(ReadOnlySpan<TDerived>)` |
 | string to ReadOnlySpan | `static ReadOnlySpan<char> MemoryExtensions.AsSpan(string)` |
 
+#### Overload resolution
+
+*Better conversion from expression* ([§12.6.4.5][better-conversion-from-expression]) is updated to prefer implicit span conversions.
+This is based on [collection expressions overload resolution changes][ce-or].
+
+> Given an implicit conversion `C₁` that converts from an expression `E` to a type `T₁`, and an implicit conversion `C₂` that converts from an expression `E` to a type `T₂`, `C₁` is a *better conversion* than `C₂` if one of the following holds:
+>
+> - `E` is a *collection expression* and one of the following holds:
+>   - `T₁` is `System.ReadOnlySpan<E₁>`, and `T₂` is `System.Span<E₂>`, and an implicit conversion exists from `E₁` to `E₂`.
+>   - `T₁` is `System.ReadOnlySpan<E₁>` or `System.Span<E₁>`, and `T₂` is an *array_or_array_interface* with *element type* `E₂`, and an implicit conversion exists from `E₁` to `E₂`.
+>   - `T₁` is not a *span_type*, and `T₂` is not a *span_type*, and an implicit conversion exists from `T₁` to `T₂`.
+> - `E` is not a *collection expression* and one of the following holds:
+>   - `E` exactly matches `T₁` and `E` does not exactly match `T₂`
+>   - `E` exactly matches both or neither of `T₁` and `T₂`, and `T₁` is a better conversion target than `T₂`
+>   - **the rules above do not hold and `C₁` is an implicit span conversion and `C₂` is not an implicit span conversion**
```

Comment:
Let's capture this breaking change example in this spec so we don't lose it.

Reply:
Captured, thanks. But I think we should try to update some rules to avoid this break. It seems like a common scenario. Also if the BCL tried to mitigate this break, I think they would need to add a special overload like `ArrayExtensions.Contains<T>(T[], T)` which is exactly what we want to prevent with this feature.